src/Azure/Orleans.Streaming.AzureStorage/Hosting/AzureQueueStreamProviderBuilder.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using Azure.Storage.Queues;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Orleans;
+using Orleans.Configuration;
+using Orleans.Hosting;
+using Orleans.Providers;
+
+[assembly: RegisterProvider("AzureQueueStorage", "Streaming", "Silo", typeof(AzureQueueStreamProviderBuilder))]
+
+namespace Orleans.Hosting;
+
+public sealed class AzureQueueStreamProviderBuilder : IProviderBuilder<ISiloBuilder>
+{
+    public void Configure(ISiloBuilder builder, string name, IConfigurationSection configurationSection)
+    {
+        builder.AddAzureQueueStreams(name, (OptionsBuilder<AzureQueueOptions> optionsBuilder) =>
```

Comment:
I wasn't sure if this also needs to support configuring the `MessageVisibilityTimeout`. I can get that added in if so.